transit_odp/organisation/tests/csv/test_service_codes.py

```diff
@@ -121,92 +138,308 @@ def test_registered_service_code():
+    )
+
+    # Testing something that IS in BODS but not in OTC
+    licence_number = "PD0000055"
+
+    bods_licence = BODSLicenceFactory(organisation=org1, number=licence_number)
+    dataset8 = DatasetFactory(organisation=org1)
+    TXCFileAttributesFactory(
+        revision=dataset8.live_revision,
+        operating_period_end_date=None,
+        service_code="Z10",
+        modification_datetime=datetime.datetime(2023, 2, 24),
+    )
+    SeasonalServiceFactory(
+        licence=bods_licence,
+        registration_code=service_codes[0][-1:],
+        start=datetime.datetime(2022, 2, 24),
+        end=datetime.datetime(2024, 2, 24),
+    )
 
     service_codes_csv = ServiceCodesCSV(org1.id)
-    _ = service_codes_csv.to_csv()
-
-    assert mock_writer.writerow.call_count == 1 + num_otc_services
-    call_args_list = mock_writer.writerow.call_args_list
-    assert call_args_list[0] == call(
-        ("Service Status", "Licence Number", "Service Code", "Line", "Dataset ID")
-    )
-    expected_missing_rows = [
-        call(
-            (
-                ServiceCodesCSV.MISSING_DATA,
-                licence_number,
-                service_codes[1],
-                service_numbers[1],
-                "",
-            )
-        ),
-        call(
-            (
-                ServiceCodesCSV.MISSING_DATA,
-                licence_number,
-                service_codes[2],
-                service_numbers[2],
-                "",
-            )
-        ),
+    csv_string = service_codes_csv.to_string()
+    csv_output = get_csv_output(csv_string)
+
+    assert csv_output["row0"] == [
+        '"Requires Attention"',
+        '"Published Status"',
+        '"OTC status"',
+        '"Scope Status"',
+        '"Seasonal Status"',
+        '"Staleness Status"',
+        '"Licence Number"',
+        '"Service Code"',
+        '"Line"',
+        '"Dataset ID"',
+        '"Effective Seasonal Start Date"',
+        '"Seasonal Start Date"',
+        '"Seasonal End Date"',
+        '"Effective stale date due to end date"',
+        '"Effective stale date due to Effective last modified date"',
+        '"Last modified date < Effective stale date due to OTC effective date"',
+        '"Effective stale date due to OTC effective date"',
     ]
-    assert call_args_list[1] in expected_missing_rows
-    assert call_args_list[2] in expected_missing_rows
-    assert call_args_list[3] == call(
-        (
-            ServiceCodesCSV.REGISTERED,
-            licence_number,
-            service_codes[0],
-            service_numbers[0],
-            str(dataset1.id),
-        )
-    )
+
+    assert csv_output["row1"][0] == '"No"'
+    assert csv_output["row1"][1] == '"Unpublished"'
+    assert csv_output["row1"][2] == '"Registered"'
+    assert csv_output["row1"][3] == '"In Scope"'
+    assert csv_output["row1"][4] == '"Out of Season"'
+    assert csv_output["row1"][5] == '"Not Stale"'
+    assert csv_output["row1"][14] == '""'
```

Comment:
It seems arbitrary that you're testing the output of columns 0-5, 14-16 in all cases. In several of those cases, you don't care about columns 14-16 as they're blank.

Can you add some asserts that cover the other columns, where it's appropriate, please - at least the seasonal start and end dates, and "Effective stale date due to end date". Only where it's relevant, not in every row!